core/shared/src/main/scala/zio/Ref.scala

```diff
@@ -135,6 +135,49 @@ abstract class Ref[A] extends Serializable {
       val result = pf.applyOrElse[A, A](v, identity)
       (result, result)
     }
+
+  /**
+   * Atomically increments the current value of the `Ref` by 1, returning the
+   * value immediately before modification.
+   */
+  final def getAndIncrement(implicit num: math.Numeric[A], trace: Trace): UIO[A] =
```

Comment:
Is it wise to mark them as final? I suppose a specialized version won't have the implicit argument anyway, so maybe.

Reply:
Making them non-final is fully backwards compatible, so maybe we can start with having them as final and we can change them when needed